src/core/net/dns_headers.cpp

```diff
@@ -35,13 +35,70 @@
 
 #include "common/code_utils.hpp"
 #include "common/debug.hpp"
+#include "common/random.hpp"
 #include "common/string.hpp"
 
 namespace ot {
 namespace Dns {
 
 using ot::Encoding::BigEndian::HostSwap16;
 
+otError Header::SetRandomMessageId(void)
+{
+    return Random::Crypto::FillBuffer(reinterpret_cast<uint8_t *>(&mMessageId), sizeof(mMessageId));
```

Comment:
```suggestion
    return Random::NonCrypto::FillBuffer(reinterpret_cast<uint8_t *>(&mMessageId), sizeof(mMessageId));
```
nit: Is `NonCrypto` secure enough?

Reply:
I don't think `NonCrypto` is sufficient, since the sequence is predictable.

From https://tools.cisco.com/security/center/resources/dns_best_practices:

> If attackers are able to predict the next transaction ID used in the DNS query along with source port value, they can construct and send (spoof) DNS messages with the correct transaction ID. Even though the DNS message sent by the attacker is falsified, the DNS resolver accepts the query response because the transaction ID and source port value match up with the query the resolver sent, resulting in the DNS resolvers cache being poisoned.

Reply:
Good catch! Thanks. I think I wanted it to be crypto as well and it was a typo :)